src/main/scala/battleship/GameActor.scala

```diff
@@ -54,37 +54,37 @@ class GameActor extends Actor with ActorLogging {
     else one
   }
 
-
   override def receive: Receive = {
     case StartGame(size, player1, player2, boatSet) => {
       player1 ! PlaceBoats(boatSet, size)
+      player2 ! PlaceBoats(boatSet, size)
       sender() ! "Game started!"
-      context.become(gameInit1(size, player1, player2, boatSet))
+      context.become(gameInit(size, player1, player2))
     }
     case GameStateRequest => sender() ! GameNotStartedYet(Uninitialised)
   }
 
-  def gameInit1(size: Int, player1: ActorRef, player2: ActorRef, boatset: Seq[Boat]): Receive = {
-    case BoatSetup(placement1) if sender() == player1 =>
-      player2 ! PlaceBoats(boatset, size)
-      context.become(gameInit2(sender(), size, player1, player2, placement1))
-    case GameStateRequest => sender() ! GameNotStartedYet(Player1AskedForBoats)
-  }
-
-  def gameInit2(gameStarter: ActorRef, size: Int, player1: ActorRef, player2: ActorRef, placement1: Set[(Boat, BoatLocation)]): Receive = {
-    case BoatSetup(placement2) if sender() == player2 =>
-      val boardStates = Map(player1 -> placeBoats(placement2), player2 -> placeBoats(placement1))
-      player2 ! GetNextShot(size, boardStates(player2).history)
-      context.become(gameStarted(gameStarter, size, player1, player2, player2, boardStates))
-    case GameStateRequest => sender() ! GameNotStartedYet(Player2AskedForBoats)
+  def gameInit(size: Int, player1: ActorRef, player2: ActorRef, boardStates: BoardStates=Map.empty): Receive = {
+    case BoatSetup(placement) if (sender() == player1 || sender() == player2) =>
+      val nextBoardStates = boardStates + (sender() -> placeBoats(placement))
+      sender() ! ReceivedPlacement(BoatSetup(placement))

```

Comment:
`ReceivedPlacement` is not really a message name. Maybe we should refactor this to a separate message case class.